Project-18-Food-order-app/src/services/stores/CartContext.jsx

```diff
@@ -0,0 +1,99 @@
+import { useState, createContext } from "react";
+import PropTypes from "prop-types";
+import { URL } from "../constants/constants";
+import useHttp from "../../hooks/useHttp";
+
+const requestConfig = {
+  method: "GET",
+};
+
+const CartContext = createContext({
+  mealData: [],
+  cartData: [],
+  isLoading: false,
+  error: "",
+  clearCartData: () => {},
+  onAddItem: () => {},
+  onRemoveItem: () => {},
+});
+
+const CartContextProvider = ({ children }) => {
+  const [cartData, setCartData] = useState([]);
+
+  const {
+    data: mealData,
+    isLoading,
+    error,
+  } = useHttp(`${URL}meals`, requestConfig, []);
```

Comment:
@sknepali Evaluate the performance impact of fetching mealData on every render. Consider memoization or optimizing the data fetching process, depending on how often it needs to change.

Reply:
Here, useHttp won't fetch data again because in useHttp I wrap the logic to make http request inside useCallback which will be called only when any one of its dependency will change.